### DISCUSSION TITLE
How to start contributing?

### DISCUSSION THREAD

**Discussion by drpaneas** [hajimehoshi/ebiten#1616] [Q&A] [answered]
First off, let me start by expressing my gratitude for the creating and sharing Ebiten with the rest of us. It's such an amazing idea and open gesture.

Personally I think I find myself more interested in creating tools for making games, rather than making games themselves.

Also, in my daily job I work a lot with Go and I really like it.

Combining graphics and Go, seems a very natural to my interest. That has been told, I would like to hear your opinion how one can start getting familiar with codebase of Ebiten. Is there any good place to start?

My approach would be to write a simple helloworld program and start debugging it, in order to see all the steps delve is going to step into.

Any ideas, proposals and comments are welcome :)

**Comment by hajimehoshi** [comment, marked as answer]
Hi, thank you for taking a look at Ebiten!

To know what Ebiten's implementation does, there are few documents. For example,
https://docs.google.com/document/d/1m60UWG9LFWgpO1vQaf5cUPx3MIsyRGklbOkT2mPxoAo/edit?usp=sharing

To know what to contribute, there are some useful labels:
Good-first issues: https://github.com/hajimehoshi/ebiten/issues?q=is%3Aopen+is%3Aissue+label%3Agood-first-issue
Help-wanted: https://github.com/hajimehoshi/ebiten/issues?q=is%3Aopen+is%3Aissue+label%3Ahelp-wanted
Help-wanted issues might be technically difficult.

To know how to contribute, see https://github.com/hajimehoshi/ebiten/blob/main/CONTRIBUTING.md

> My approach would be to write a simple helloworld program and start debugging it, in order to see all the steps delve is going to step into.

This sounds good :-) Probably causing panicking on purpose shows the stacktrace, which is very informative. I'm afraid there are very small number of documents explaining the implementation details about Ebiten. Feel free to ask question about Ebiten's usage and/or implementation any time.

**Reply by drpaneas** [reply]
Thank you for your fast reply. Is this (Github) a good place to place questions or you would prefer to use Slack?

**Reply by hajimehoshi** [reply]
Both GitHub discussion and Slack are fine!
